catkin_ws/src/state_estimation/src/sensors.py

```diff
@@ -19,111 +18,121 @@
 
 
 class Sensor():
-    def __init__(self, sensor_name):
-        self.time_before_considered_inactive = 1 #seconds
-        self.last_error_message_time = rospy.get_time()
+    def __init__(self, sensor_name, is_active=0):
+        self.is_active = is_active
         self.sensor_name = sensor_name
 
-        self.x = 0
-        self.y = 0
-        self.z = 0
-        
-        self.q_nwu_auv = np.quaternion(1, 0, 0, 0)
-        self.angular_velocity = np.array([0,0,0])
+    def get_sensor_name(self):
+        return self.sensor_name
 
-        # initialize a sensor as "inactive"
-        self.last_unique_state_time = -1 * self.time_before_considered_inactive
-        self.last_state = [self.x,self.y,self.z,self.q_nwu_auv,self.angular_velocity]
+    def sensor_status_cb(self, msg):
+        self.is_active = msg.data
+
+    # @abstractmethod
+    def get_reading(self):
+        raise NotImplementedError("Subclass must implement abstract method")
     
-    def updateLastState(self):
-        current_state = [self.x,self.y,self.z,self.q_nwu_auv,self.angular_velocity]
-        different = False
-        for i in range(len(current_state)):	
-            if isinstance(current_state[i], Iterable):
-                different = True if different else any([curr_state_el != last_state_el for curr_state_el, last_state_el in zip(current_state[i], self.last_state[i])])
-            else:
-                different = True if different else current_state[i] is not self.last_state[i]
-        
-        if different:
-            if rospy.get_time() - self.last_unique_state_time > self.time_before_considered_inactive:
-                rospy.loginfo("{} has become active.".format(self.sensor_name))
-            self.last_unique_state_time = rospy.get_time()
-            self.last_state = current_state
-        
-    def isActive(self):
-        if rospy.get_time() == 0: return False
-        if not self.hasValidData(): return False #check that state is complete
-        if rospy.get_time() - self.last_unique_state_time > self.time_before_considered_inactive: #check that state has changed in last N seconds
-            if rospy.get_time() - self.last_error_message_time > 1:
-                self.last_error_message_time = rospy.get_time()
-                rospy.logwarn("{} has been inactive for {} seconds.".format(self.sensor_name, self.time_before_considered_inactive))
-            return False
-        else:
-            return True
```

Comment:
the warning messages were all moved to the sensor status, and I also reduced how frequent they were because they are annoying as fuck in the sim. When you are not moving the auv in the sim, the sensors readings are exactly the same. So, the status check thinks that they stopped working and starts spamming warning messages